diff --git a/AMBuildScript b/AMBuildScript
index 81a117d..2d84a83 100644
--- a/AMBuildScript
+++ b/AMBuildScript
@@ -169,10 +169,11 @@ class ExtensionConfig(object):
       '-pipe',
       '-fno-strict-aliasing',
       '-Wall',
-      '-Werror',
+      #'-Werror',
       '-Wno-unused',
       '-Wno-switch',
       '-Wno-array-bounds',
+      '-Wno-class-memaccess',
       '-msse',
       '-m32',
       '-fvisibility=hidden',
