diff --git a/AMBuildScript b/AMBuildScript
index c173ac9..0af7a16 100644
--- a/AMBuildScript
+++ b/AMBuildScript
@@ -186,7 +186,7 @@ class ExtensionConfig(object):
       '-pipe',
       '-fno-strict-aliasing',
       '-Wall',
-      '-Werror',
+      #'-Werror',
       '-Wno-unused',
       '-Wno-switch',
       '-Wno-array-bounds',
@@ -195,7 +195,7 @@ class ExtensionConfig(object):
       '-fvisibility=hidden',
     ]
     cxx.cxxflags += [
-      '-std=c++11',
+      '-std=c++14',
       '-fno-exceptions',
       '-fno-threadsafe-statics',
       '-Wno-non-virtual-dtor',
@@ -468,7 +468,7 @@ BuildScripts = [
 
 if builder.backend == 'amb2':
   BuildScripts += [
-    'scripting/AMBuilder',
+    #'scripting/AMBuilder',
     'PackageScript',
   ]
 
diff --git a/RegNatives.cpp b/RegNatives.cpp
index fa6e53d..9402465 100644
--- a/RegNatives.cpp
+++ b/RegNatives.cpp
@@ -4,11 +4,11 @@
 RegNatives g_RegNatives;
 
 void RegNatives::Register(ICallWrapper *pWrapper) {
-  m_Natives.append(pWrapper);
+  m_Natives.emplace_back(pWrapper);
 }
 
 void RegNatives::UnregisterAll() {
-  for (size_t iter = 0; iter < m_Natives.length(); ++iter) {
+  for (size_t iter = 0; iter < m_Natives.size(); ++iter) {
     m_Natives[iter]->Destroy();
   }
 
diff --git a/RegNatives.h b/RegNatives.h
index 4a44367..b4b782b 100644
--- a/RegNatives.h
+++ b/RegNatives.h
@@ -8,7 +8,7 @@ public:
   void Register(ICallWrapper *pWrapper);
   void UnregisterAll();
 private:
-  ke::Vector<ICallWrapper *> m_Natives;
+  std::vector<ICallWrapper *> m_Natives;
 };
 
 extern RegNatives g_RegNatives;
