diff --git a/AMBuildScript b/AMBuildScript
index 839b733..4b3ad72 100644
--- a/AMBuildScript
+++ b/AMBuildScript
@@ -267,6 +267,9 @@ class SteamWorksConfig(object):
       os.path.join(self.sm_root, 'public'),
       os.path.join(self.sm_root, 'public', 'extensions'),
       os.path.join(self.sm_root, 'public', 'sourcepawn'),
+      os.path.join(self.sm_root, 'public', 'amtl'),
+      os.path.join(self.sm_root, 'public', 'amtl', 'amtl'),
+      os.path.join(self.sm_root, 'sourcepawn', 'include'),
     ]
     return compiler
 	
